user-impl/src/main/java/com/example/auction/user/impl/UserServiceImpl.java

```diff
@@ -45,14 +49,42 @@ public UserServiceImpl(PersistentEntityRegistry registry, ActorSystem system, Ma
     }
 
     @Override
+    public ServiceCall<Credential, Done> updateCredential() {
+        return credential -> {
+            String hashed = BCrypt.hashpw(credential.getPassword(), BCrypt.gensalt());
+            return credRef(credential.getUsername()).ask(new CredentialCommand.UpdateCredential(credential.getId(), credential.getUsername(), hashed))
+               .thenApply(maybeCredential -> Done.getInstance());
+        };
+    }
+
+    @Override
     public ServiceCall<NotUsed, User> getUser(UUID userId) {
         return req -> {
             return entityRef(userId).ask(UserCommand.GetUser.INSTANCE)
                     .thenApply(maybeUser ->
                             maybeUser.orElseGet(() -> {
                                 throw new NotFound("User " + userId + " not found");
                             })
-            );
+                    );
+        };
+    }
+
+    @Override
+    public ServiceCall<Credential, String> login() {
+        return (Credential req) -> {
+            return credRef(req.getUsername()).ask(CredentialCommand.GetCredential.INSTANCE)
```

Comment:
Perhaps rather than doing this logic here, instead create a `Login` command instead of a `GetCredential` command. Then, the login business logic can be done in the persistent entity.

Then again, someone may disagree with me, @ignasi35 or @TimMoore what do you think?